### DISCUSSION TITLE
have a way to exit fullscreen mode

### DISCUSSION THREAD

**Discussion by KiaraGrouwstra** [YaLTeR/niri#337] [Ideas]
right now, we have an action to toggle full-screen mode, which i sometimes use.
i also have keys/scripts using spawn actions, and sometimes use these when i am in full-screen mode.

under these circumstances the present window stays in full-screen, which occasionally i experience as counter-intuitive, as seemingly nothing happened.
i would like to have these instead have niri exit full-screen if applicable. however, it doesn't seem there is currently a way to do this: the existing action toggles, whereas there is no current way check whether one is in full-screen mode (i think?) or to simply exit full-screen rather than toggling.

it would be nice if we could add either of those to facilitate this use-case.

**Comment by YaLTeR** [comment]
That makes sense. Guess we can have a... uh, looks like I already named this `fullscreen-window` and not `toggle-fullscreen-window`... uhm. What should we call it? `fullscreen-window fullscreen=true/false` maybe? I'm kind of out of good ideas on this one.

**Reply by KiaraGrouwstra** [reply]
works for me! 😄

**Reply by Dlurak** [reply]
How about `fullscreen-window fullscreen=fullscreen/normal/toggle`?

**Reply by Dlurak** [reply]
I've opened the pr #1968 for it in a way that is downwards compatible :)

**Comment by darkliquid** [comment]
To further add to this, there doesn't appear to be a way to determine if a window is actually fullscreen or not (i.e. there is no attribute like `is-floating`, etc that can be looked for via ipc. Having this ability would allow third party tools to trigger stuff based on fullscreen state.

My specific use case for that is that I'd like to have a workspace that all fullscreen windows get moved to, which without a window matcher for window rules, nor an ipc method of determining fullscreen state, I can't automatically do.
